packages/storage/src/core/storage.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  extractObjectFiles,
+  replaceObjectFilesWithUris,
+  replaceObjectGatewayUrlsWithSchemes,
+  replaceObjectSchemesWithGatewayUrls,
+} from "../common/utils";
+import {
+  FileOrBuffer,
+  FileOrBufferSchema,
+  IpfsUploadBatchOptions,
+  IStorageDownloader,
+  IStorageUploader,
+  Json,
+  UploadOptions,
+} from "../types";
+import { StorageDownloader } from "./downloaders/storage-downloader";
+import { IpfsUploader } from "./uploaders/ipfs-uploader";
+
+export class ThirdwebStorage<T extends UploadOptions = IpfsUploadBatchOptions> {
```

Comment:
make sure to add some docs / examples

Reply:
Indeed - will put these in separate PR like Solana